pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -514,6 +514,9 @@ type CommonPrometheusFields struct {
 	// It is meant to be used by admins to keep the overall number of
 	// samples/series under a desired limit.
 	//
+	// Starting from Prometheus v2.45.0, this field is used to define global limits if `spec.sampleLimit` is not set.
+	// Admins can combine this field with `spec.sampleLimit` to provide a global upper limit and a lower limit for individual objects.
```

Comment:
It's still a bit unclear IMHO. What if we explain in details the decision tree?

```
When both `enforcedSampleLimit` and `sampleLimit` are defined and greater than zero, the following rules apply:
* Scrape objets without a defined sampleLimit value will inherit the global sampleLimit value (Prometheus <= 2.45.0) or the enforcedSampleLimit value (Prometheus > v2.45.0).
* Scrape objets with a sampleLimit value less than or equal to enforcedSampleLimit keep their specific value.
* Scrape objets with a sampleLimit value greater than enforcedSampleLimit are set to enforcedSampleLimit.
```

From my POV, it's ok to have the detailed documentation for 1 field (e.g. enforced sample limit) and have a reference (e.g. "See `enforcedSampleLimit` to understand the interactions with `<field>`."). It might also good to have a reference to the enforced fields in each global field.